server/sonar-server/src/main/java/org/sonar/server/rule/RegisterRules.java

```diff
@@ -572,4 +576,45 @@ private void update(DbSession session, RuleDefinitionDto rule) {
     dbClient.ruleDao().update(session, rule);
   }
 
+
+  private static void verifyRuleKeyConsistency(RulesDefinition.Context context) {
+    List<RulesDefinition.Rule> allRules = context.repositories().stream()
```

Comment:
you should use the repositories retrieved at line 117. This method will miss rules defined in extended repositories.